### PR TITLE
Upgrade Mage to Ubuntu 24.04

### PR DESCRIPTION
**Pull request by DavIvek**
### Description

We need to upgrade from Debian 11 to Ubuntu 24.04 to accommodate Memgraph binaries built with toolchain v6 and to ensure security support with Ubuntu 24.04. 
This PR is blocked by https://github.com/memgraph/memgraph/pull/2638.

### Pull request type

- [ ] Bugfix
- [ ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Related issues

Delete if this PR doesn't resolve any issues. Link the issue if it does.

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Tests provided (unit / e2e)
- [ ] Code documentation
- [ ] README short description

### Documentation checklist
- [ ] Add the documentation label tag
- [ ] Add the bug / feature label tag
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[Release note text]**
- [ ] Link the documentation PR here
    - **[Documentation PR link]**
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: Dockerfile.no_ML [hunk 1]
```diff
@@ -1,43 +1,47 @@
...
     && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 COPY memgraph-${TARGETARCH}.deb .
 
-RUN dpkg -i memgraph-${TARGETARCH}.deb && rm memgraph-${TARGETARCH}.deb && rm /usr/lib/memgraph/query_modules/schema.so  \
-&& rm /usr/lib/memgraph/query_modules/example_cpp.so
+# Install Memgraph and remove unwanted modules
+RUN dpkg -i memgraph-${TARGETARCH}.deb && rm memgraph-${TARGETARCH}.deb \
+    && rm /usr/lib/memgraph/query_modules/schema.so \
+    && rm /usr/lib/memgraph/query_modules/example_cpp.so
```
Comment (gitbuda) [inline Dockerfile.no_ML:44]: What's the issue here? Why are these modules "unwanted"?

File: python/requirements.txt [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-six==1.16.0
-torchmetrics==0.9.3
-igraph==0.10.2
-scikit-learn==0.24.2
 gqlalchemy==1.4.1
+igraph==0.10.2
 mysql-connector-python==8.0.32
-oracledb==1.2.2
-pyodbc==4.0.35
+networkx==2.6.2
```
Comment (gitbuda) [inline python/requirements.txt:8]: E.g. NetworkX latest version is 3.4.2 -> go over this list again, and try to bump everything

The only concern I have are these database connector like `mysql-connector-python` because that might be sensitive to the version of databases our customers are using... 🤔 But, I would still bump that, if there is a problem for a given customer we can manually downgrade directly inside the Docker image.

File: Dockerfile.no_ML (hunk 1 above)
Comment (DavIvek) [reply Dockerfile.no_ML:44]: This [issue](https://github.com/memgraph/memgraph/issues/1559) might be related, but since it is closed, we can likely reintroduce them

Comment (DavIvek) [reply Dockerfile.no_ML:44]: Maybe we should remove `schema` module from `mage` since we have it in `memgraph` already.

Comment (gitbuda) [reply Dockerfile.no_ML:41]: But this will delete the memgraph one because that one comes installed with Memgraph during the install, we should just delete the one under mage if it's the same. To start with, just skip installing the one from mage 🤔

File: python/kmeans.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ def get_clusters(
     n_init: mgp.Number = 10,
     max_iter: mgp.Number = 10,
     tol: mgp.Number = 1e-4,
-    algorithm: str = "auto",
+    algorithm: str = "lloyd",
```
Comment (DavIvek) [inline python/kmeans.py:56]: `scikit-learn`: changed in version 1.1: Renamed “full” to “lloyd”, and deprecated “auto” and “full”. Changed “auto” to use “lloyd” instead of “elkan” (https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html)
This might be `breaking` for some users + [documentation](https://memgraph.com/docs/advanced-algorithms/available-algorithms/kmeans_clustering) needs to be updated.

**Comment by DavIvek** [comment]
This PR is blocked by https://github.com/memgraph/memgraph/pull/2638.

File: e2e/kmeans_test/test_kmeans_small/test.yml
Comment (DavIvek) [inline e2e/kmeans_test/test_kmeans_small/test.yml:1]: Due to the newer version of `scikit-learn`, the `cluster_ids` have changed, but the clusters themselves remain the same.

**Comment by DavIvek** [comment]
TODO:
Update tests to use `ubuntu 24.04.`

File: python/kmeans.py (hunk 3 above)
Comment (gitbuda) [reply python/kmeans.py:56]: I agree -> added the breaking label to the PR

**Review by gitbuda — APPROVED** [review]
